From 0689af244a331e98fa2a82459e0d6e1b22b193d6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 18 Jul 2021 16:49:56 +0300
Subject: [PATCH] [SW-0] gaudi2: include headers explicitly

Do not rely on "header to include another header". Include the
gaudi2_blocks_linux_driver.h header explicitly where it needs to be
used.

Change-Id: I101eb0f6cd5dc45af4094a1d5ecfb31db6749180
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index f49b74c..928a32e 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -5,6 +5,7 @@
  *
  */
 
+#include <gaudi2_blocks_linux_driver.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
 
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 88190ce..a208da6 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -6,7 +6,6 @@
 #ifndef COMMON_GAUDI2_ARC_GNU_H_
 #define COMMON_GAUDI2_ARC_GNU_H_
 
-#include <gaudi2_blocks_linux_driver.h>
 #include <bitops.h>
 
 #define SRAM_SIZE			0x3000000	/* 48MB */
diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index 41c5890..517b76f 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -5,6 +5,7 @@
  *
  */
 
+#include <gaudi2_blocks_linux_driver.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
 
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 119f823..cdd9f5a 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -5,6 +5,7 @@
  *
  */
 
+#include <gaudi2_blocks_linux_driver.h>
 #include <psoc_gpio1_regs.h>
 #include <psoc_qspi_regs.h>
 #include <psoc_efuse_regs.h>
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index 53c1868..1820ac6 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -6,7 +6,6 @@
 #ifndef COMMON_GAUDI2_ARM_H_
 #define COMMON_GAUDI2_ARM_H_
 
-#include <gaudi2_blocks_linux_driver.h>
 #include <bitops.h>
 
 #ifdef TARGET_IS_ONIX
